x-pack/test/functional/services/machine_learning/data_visualizer_file_based.ts

```diff
@@ -0,0 +1,110 @@
+import { MlCommon } from './common';
+
+export function MachineLearningDataVisualizerFileBasedProvider(
+  { getService, getPageObjects }: FtrProviderContext,
+  mlCommon: MlCommon
+) {
+  const log = getService('log');
+  const retry = getService('retry');
+  const testSubjects = getService('testSubjects');
+
+  const PageObjects = getPageObjects(['common']);
+
+  return {
+    async selectFile(path: string, expectError: boolean = false) {
+      log.debug(`Importing file '${path}' ...`);
+      await PageObjects.common.setFileInputPath(path);
+
+      await testSubjects.waitForDeleted('mlPageFileDataVisLoading');
+
+      if (expectError) {
+        await testSubjects.existOrFail('~mlFileUploadErrorCallout');
+      } else {
+        await testSubjects.missingOrFail('~mlFileUploadErrorCallout');
+        await testSubjects.existOrFail('mlPageFileDataVisResults');
+      }
+    },
+
+    async assertFileTitle(expectedTitle: string) {
+      const actualTitle = await testSubjects.getVisibleText('mlFileDataVisResultsTitle');
+      expect(actualTitle).to.eql(
+        expectedTitle,
+        `Expected file title to be '${expectedTitle}' (got '${actualTitle}')`
+      );
+    },
+
+    async assertFileContentPanelExists() {
+      await testSubjects.existOrFail('mlFileDataVisFileContentPanel');
+    },
+
+    async assertSummaryPanelExists() {
+      await testSubjects.existOrFail('mlFileDataVisSummaryPanel');
+    },
+
+    async assertFileStatsPanelExists() {
+      await testSubjects.existOrFail('mlFileDataVisFileStatsPanel');
+    },
+
+    async navigateToFileImport() {
+      await testSubjects.click('mlFileDataVisOpenImportPageButton');
+      await testSubjects.existOrFail('mlPageFileDataVisImport');
+    },
+
+    async assertImportSettingsPanelExists() {
+      await testSubjects.existOrFail('mlFileDataVisImportSettingsPanel');
+    },
+
+    async assertIndexNameValue(expectedValue: string) {
+      const actualIndexName = await testSubjects.getAttribute(
+        'mlFileDataVisIndexNameInput',
+        'value'
+      );
+      expect(actualIndexName).to.eql(
+        expectedValue,
+        `Expected index name to be '${expectedValue}' (got '${actualIndexName}')`
+      );
+    },
+
+    async setIndexName(indexName: string) {
+      await mlCommon.setValueWithChecks('mlFileDataVisIndexNameInput', indexName, {
+        clearWithKeyboard: true,
+      });
+      await this.assertIndexNameValue(indexName);
+    },
+
+    async assertCreateIndexPatternCheckboxValue(expectedValue: boolean) {
+      const isChecked = await testSubjects.isChecked('mlFileDataVistCreateIndexPatternCheckbox');
+      expect(isChecked).to.eql(
+        expectedValue,
+        `Expected create index pattern checkbox to be ${expectedValue ? 'checked' : 'unchecked'}`
+      );
+    },
+
+    async setCreateIndexPatternCheckboxState(newState: boolean) {
+      const isChecked = await testSubjects.isChecked('mlFileDataVistCreateIndexPatternCheckbox');
```

Comment:
And again, should just be `mlFileDataVisCreateIndexPatternCheckbox`

Reply:
Fixed in a0d3fef